golem-base-indexer/golem-base-indexer-logic/src/repository/timeseries/block_transactions.rs

```diff
@@ -0,0 +1,60 @@
+use anyhow::{Context, Result};
+use sea_orm::{prelude::*, FromQueryResult, QueryOrder, QuerySelect};
+use tracing::instrument;
+
+use crate::types::{BlockTransactionPoint, ChartInfo};
+use golem_base_indexer_entity::transactions;
+
+#[derive(Debug, FromQueryResult)]
+struct DbChartBlockTransactions {
+    pub block_number: i32,
+    pub tx_count: i64,
+}
+
+const DEFAULT_BLOCK_LIMIT: u64 = 100;
+
+#[instrument(skip(db))]
+pub async fn timeseries_block_transactions<T: ConnectionTrait>(
+    db: &T,
+) -> Result<(Vec<BlockTransactionPoint>, ChartInfo)> {
+    let results = transactions::Entity::find()
+        .select_only()
+        .column(transactions::Column::BlockNumber)
+        .column_as(transactions::Column::BlockNumber.count(), "tx_count")
+        .filter(transactions::Column::BlockNumber.is_not_null())
+        .filter(transactions::Column::BlockConsensus.eq(true))
+        .group_by(transactions::Column::BlockNumber)
+        .order_by_desc(transactions::Column::BlockNumber)
+        .limit(DEFAULT_BLOCK_LIMIT)
+        .into_model::<DbChartBlockTransactions>()
+        .all(db)
+        .await
+        .context("Failed to get block transactions timeseries")?;
+
+    let chart = generate_points_block_transactions(results)?;
+
+    let info = ChartInfo {
+        id: "blockTransactions".to_string(),
+        title: "Block Transactions".to_string(),
+        description: "Number of transactions for recent blocks".to_string(),
+    };
+
+    Ok((chart, info))
+}
+
+fn generate_points_block_transactions(
+    mut db_results: Vec<DbChartBlockTransactions>,
+) -> Result<Vec<BlockTransactionPoint>> {
+    // Get ASC order for the chart
+    db_results.reverse();
```

Comment:
Wouldn't it be easier to just `order_by_asc` in the query?

Reply:
However, we need the last 100 blocks, so the query must remain in descending order.

Reply:
Ah, right, you're correct. Yup, this is the easiest way.